CONTRIBUTING.md

```diff
@@ -0,0 +1,231 @@
+
+We use the [google docstring format](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).
+
+_Not all parts of `synadm` are populated with such docstrings yet but we require it for newly written code._
+
+### Code Style
+
+We try to follow Python's [PEP 8](https://www.python.org/dev/peps/pep-0008/) guidelines as good as possible. We have an
+automatic linter in place in our CI pipeline which runs whenever code is pushed to a branch. You'll notice a check going green or red at the very bottom when viewing your open PR. The check is called "Lint with flake8". If after a `git push` you notice the check gone red, click on it and you'll see details of what the linter detected you should correct.
+
+`flake8` can be used directly on your code as well. Install it to your dev-environment:
+
+```
+pip install flake8
+```
+
+Run with:
+
+`flake8 the_file_being_worked_on.py`
+
+or leave out the filename to run it on all files and subdirectories.
+
+Some more style and coding requirements:
+
+- Pythonism: `synadm` is a Python project and as such tries to use some of the very handy features of the language. Some  of those might not be familiar to you since often we realize that contributors are professional programmers used to other languages or even are admins who are rather at home on the shell than in Python code. That is certainly all fine and we are trying to support you with refactoring your code to become more pythonic.
+- That said, we would still like `synadm` to stay approachable to Python beginners and would encourage contributors to try not to overcomplicate things.
+- We require a line length of 79 characters in Python code.
+  - Some editors have handy commands to quickly reformat multi-line text (e.g help texts of options). For example in `vim`, lines can be marked using `shift+V` and then reformatted by typing `gq`
+  - For code passages, staying within this boundary and at the same time keeping code well readable, this can be achieved using PEP8's many [line continuation suggestions](https://peps.python.org/pep-0008/#code-lay-out).
+- We don't require a maximum line length in Markdown files, for example _this document_ or `README.md`. Please omit line-breaks in sentences, paragraphs, and so on, when editing these docs.
+- If you'd like to tidy up your commits or reduce commit count we welcome the use of `git commit --amend` and `git rebase`. If you'd like us to test your code or work together within your feature branch, be aware of that force pushing might complicate things and should rather be omitted in that case.
+
+
+## Maintainers Notes
+
+This section is a checklist for maintainers of the `synadm` project. Still there might be helpful information applying to contributors as well, thus we made it public.
```

Comment:
"made it public" or "kept it public"?

```suggestion
This section is a checklist for maintainers of the `synadm` project. This information may be useful for contributors, so we kept it public.
```

Reply:
I don't know. kept it sounds good. just a feeling